src/controllers/v1/transaction/index.ts

```diff
@@ -53,10 +68,27 @@ export class TransactionController extends Controller {
    */
   @Response<NotFound>(404, 'Item not found')
   @Get('{transactionId}')
-  public async getTransaction(@Path() transactionId: UUID): Promise<TransactionResponse> {
+  public async getTransaction(
+    @Request() req: ScopedRequest,
+    @Path() transactionId: UUID
+  ): Promise<TransactionResponse> {
     const [transaction] = await this.db.getTransaction(transactionId)
     if (!transaction) throw new NotFound('transaction')
 
+    const grantedApiTypes = grantedApiTypesFromScopes(req)
+    if (!grantedApiTypes.includes(transaction.apiType)) throw new OauthError('MISSING_SCOPES')
+
     return transaction
   }
 }
+
+const grantedApiTypesFromScopes = (req: ScopedRequest, queryApiType?: TransactionApiType) => {
+  const scopes = req.user?.jwt?.scope?.split(' ') ?? []
+  return scopes.reduce<TransactionApiType[]>((acc, scope) => {
+    const scopedApiType = scopeToApiTypeMap[scope as TransactionScope]
+
+    // ignore scope if: doesn't grant access to an api type OR match the apiType in query (if provided)
+    if (!scopedApiType || (queryApiType && queryApiType !== scopedApiType)) return acc
+    return [...acc, scopedApiType]
+  }, [])
```

Comment:
Can I suggest this function doesn't take `queryApiType` as a parameter and instead in `getAllTransactions` you can just check that the `apiType` is in the returned array. The conditional on ln 91 is pretty hard to read

Reply:
done, much easier to read